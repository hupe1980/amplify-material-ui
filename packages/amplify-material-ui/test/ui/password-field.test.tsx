import * as React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import { PasswordField } from '../../src';
import { withTheme } from '../helpers/withTheme';

describe('password-field', () => {
  it('should be rendered correctly [type=password]', () => {
    const { asFragment } = render(withTheme(<PasswordField />));
    expect(asFragment()).toMatchSnapshot();
  });

  it('should change the password visibility', () => {
    const { getByTestId } = render(withTheme(<PasswordField />));

    const passwordInput = getByTestId('password-input') as HTMLInputElement;

    act(() => {
      fireEvent.click(getByTestId('togglePasswordVisibility'));
    });

    expect(passwordInput.type).toEqual('text');

    act(() => {
      fireEvent.click(getByTestId('togglePasswordVisibility'));
    });

    expect(passwordInput.type).toEqual('password');
  });
});
