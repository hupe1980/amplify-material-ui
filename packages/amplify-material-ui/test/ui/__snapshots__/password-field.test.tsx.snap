// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`password-field should be rendered correctly [type=password] 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="password-input"
        id="mui-1"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="Toggle password visibility"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          data-testid="togglePasswordVisibility"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityOffIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-nnbavb"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
